bloaty: init at 2016.11.16

dtzWill opened the pull request:
###### Motivation for this change

New tool for profiling size of binaries, from Google.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
bloaty: init at 2016.11.07

dtzWill commented:
I have not tested this on non-Linux (NixOS) platforms, but quick inspection suggests upstream supports at least Darwin as well.  LMK if I should change `meta.platforms`.

rasendubi commented on pkgs/tools/misc/bloaty/default.nix:
```diff
@@ -0,0 +1,34 @@
+  version = "2016.11.07";
+  name = "bloaty-${version}";
+
+  src = fetchgit {
```
Prefer `fetchFromGitHub` over `fetchgit`

rasendubi replied:
`fetchFromGitHub` won't work, as bloaty needs fetch submodules. Would be good to add `fetchSubmodules = true;` to `fetchgit` arguments. (Be explicit, so nobody asks why not `fetchFromGitHub`. :slightly_smiling_face:)

rasendubi commented on pkgs/tools/misc/bloaty/default.nix:
```diff
@@ -0,0 +1,34 @@
+    install -Dm755 {.,$out/bin}/bloaty
+  '';
+
+  meta = with stdenv.lib; {
```
Would you like to maintain the package?

rasendubi commented on pkgs/top-level/all-packages.nix:
```diff
@@ -489,6 +489,8 @@ in
 
   autorevision = callPackage ../tools/misc/autorevision { };
 
+  bloaty = callPackage ../tools/misc/bloaty { };
```
Better move to `development/tools`

rasendubi commented:
It would be also good to squash your commits

rasendubi commented:
@dtzWill, would you like to make the changes?

dtzWill commented:
Yes, sorry for the delay! I'll get to this today thank you! :)

dtzWill commented:
Thanks for the review, I've rebased and squashed the commits and made the mentioned changes.

Also, updated to latest git version while I was at it.

Adding myself as maintainer, here we go! :)

rasendubi commented on pkgs/development/tools/bloaty/default.nix:
```diff
@@ -0,0 +1,31 @@
+    homepage = https://github.com/google/bloaty;
+    license = licenses.asl20;
+    platforms = platforms.unix;
+    maintainer = dtzWill;
```
Should be `maintainers = [ maintainers.dtzWill ];`

dtzWill commented:
Fixed the maintainers bit, whoops and thanks! :)

rasendubi commented:
Thank you!
